RecoLocalCalo/Castor/src/RecHitCorrector.cc

```diff
@@ -114,59 +114,47 @@ RecHitCorrector::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
    iSetup.get<CastorChannelQualityRcd>().get(p);
    CastorChannelQuality* myqual = new CastorChannelQuality(*p.product());
    
-   if (!rechits.isValid()) std::cout << "No valid CastorRecHitCollection found, please check the InputLabel..." << std::endl;
+   if (!rechits.isValid()) edm::LogWarning("CastorRecHitCorrector") << "No valid CastorRecHitCollection found, please check the InputLabel...";
    
    CastorCalibrations calibrations;
    
    auto rec = std::make_unique<CastorRecHitCollection>();
    
    for (unsigned int i=0;i<rechits->size();i++) {
    	CastorRecHit rechit = (*rechits)[i];
-	//std::cout << "rechit energy = " << rechit.energy() << std::endl;
 	double fC = factor_*rechit.energy();
 	double time = rechit.time();
-	//std::cout << "rechit energy(fC) = " << fC << " time = " << time << std::endl;
 	
 	// do proper gain calibration reading the latest entries in the condDB
 	const CastorCalibrations& calibrations=conditions->getCastorCalibrations(rechit.id());
```

Comment:
```calibrations``` and ```capid``` below are only needed ```if (doInterCalib_)```
You can move their definition under that condition, e.g.:
```
      double time = rechit.time();
      double correctedenergy = factor_*rechit.energy();
      if (doInterCalib_) {
	 // do proper gain calibration reading the latest entries in the condDB
	 int capid = 0; // take some capid, gains are the same for all capid's
	 const CastorCalibrations& calibrations=conditions->getCastorCalibrations(rechit->id());
         correctedenergy *= calibrations.gain(capid);
      }

```